src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,25 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position.
```

Comment:
@stephenc Based on the original implementation:

- `new VersionNumber("1.0.0.GA.2-3").digit(3)` returns `1`
- `new VersionNumber("1.0.0.GA.2-3").digit(4)` returns `1`